packages/client/hmi-client/src/components/model/tera-model.vue

```diff
@@ -130,10 +147,66 @@ const optionsMenuItems = ref([
 	// ,{ icon: 'pi pi-trash', label: 'Remove', command: deleteModel }
 ]);
 
-// Centralize this eventually
-function updateModelContent(updatedModel: Model) {
-	model.value = updatedModel;
-	updateModel(updatedModel);
+async function updateModelContent(updatedModel: Model) {
+	await updateModel(updatedModel);
+	setTimeout(async () => {
+		await fetchModel(); // elastic search might still not update in time
+		useResourcesStore().setActiveProject(await ProjectService.get(props.project.id ?? '102', true));
+	}, 800);
+}
+
+async function updateModelName() {
+	if (model.value && !isEmpty(newName.value)) {
+		const modelClone = cloneDeep(model.value);
+		modelClone.header.name = newName.value;
+		await updateModelContent(modelClone);
+	}
+	isRenaming.value = false;
+}
+
+async function updateConfiguration(updatedConfiguration: ModelConfiguration, index: number) {
+	await updateModelConfiguration(updatedConfiguration);
+	setTimeout(async () => {
+		emit('update-model-configuration');
+		modelConfigurations.value[index] = updatedConfiguration; // Below line would be ideal but the order of the configs change after the refetch
+		// await fetchConfigurations(); // elastic search might still not update in time
+	}, 800);
+}
+
+async function addConfiguration(configuration: ModelConfiguration) {
+	if (model.value) {
+		await createModelConfiguration(
+			model.value.id,
+			`Copy of ${configuration.name}`,
+			configuration.description as string,
+			configuration.configuration
+		);
+		setTimeout(() => {
+			emit('new-model-configuration');
+			fetchConfigurations(); // elastic search might still not update in time
+		}, 800);
+	}
+}
+
+async function fetchConfigurations() {
+	if (model.value) {
+		let tempConfigurations = await getModelConfigurations(model.value.id);
+
+		// Ensure that we always have a "default config" model configuration
+		if (
+			isEmpty(tempConfigurations) ||
+			!tempConfigurations.find((d) => d.name === 'Default config')
+		) {
+			await addDefaultConfiguration(model.value);
+			tempConfigurations = await getModelConfigurations(model.value.id);
+		}
+		modelConfigurations.value = tempConfigurations;
+	}
+}
+
+async function fetchModel() {
+	model.value = await getModel(props.assetId);
+	fetchConfigurations();
 }
```

Comment:
I think you should create a new `getModelWithConfigurations()` method in the service to do all that work, it might be useful elsewhere.